File: spring-cloud-integrated-demo/product-service/src/main/java/com/github/jccode/springcloud/integrateddemo/product/service/ProductService.java
package com.github.jccode.springcloud.integrateddemo.product.service;

import com.github.jccode.springbootsample.core.repo.CrudMapper;
import com.github.jccode.springbootsample.core.service.CrudServiceImpl;
import com.github.jccode.springcloud.integrateddemo.product.model.Product;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class ProductService extends CrudServiceImpl<Product> {

    @Autowired
    public ProductService(CrudMapper<Product> mapper) {
        super(mapper);
    }
}

File: spring-cloud-integrated-demo/product-service/src/main/scala/com/github/jccode/springcloud/integrateddemo/product/service/ProductService.scala
package com.github.jccode.springcloud.integrateddemo.product.service

import com.github.jccode.springbootsample.core.service.CrudServiceImpl
import com.github.jccode.springcloud.integrateddemo.product.repo.ProductMapper
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ProductService(@Autowired override val mapper: ProductMapper) extends CrudServiceImpl(mapper){

}
